### ISSUE TITLE
Posn shifting member functions don't meet expectations

### ISSUE THREAD

**Issue by brghena** [tov/ge211#6]
The `up_by()`, `down_by()`, etc. "shifting member functions" for `Posn` construct a new `Posn` modified from the existing one as specified. I've found that this doesn't match student's expectations that they will modify the `Posn` they are called on.

It also doesn't match my own expectations. If they were named `from_up_by()` similarly to the functions of `Rect`, I think that they would make more sense. Or if they actually modified they `Posn` they are called on.

Is it intentional behavior that the shifting member functions create a new `Posn` rather than modifying the one they are called on?

**Comment by tov** [comment]
Yes, this was the intended behavior. I think the intent was for them to get some practice with a non-mutating API like this. But if you think it’s too confusing, I would be open to one of a few different changes:

 1. We could rename the functions from `up_by` to `above_by`.

 2. We could change the signatures from
     
    ```cxx
        Posn up_by(Coord) const;
    ```
    
    to

    ```cxx
        Posn& up_by(Coord);
    ```

 3. We could get rid of them entirely, since they’re mostly redundant with `Posn Posn::operator+(Dims) const` and `Posn& Posn::operator+(Dims)`.

What do you think?
